packages/typescript-estree/src/parseSettings/warnAboutTSVersion.ts

```diff
@@ -23,23 +23,32 @@ const isRunningSupportedTypeScriptVersion = semver.satisfies(
 
 let warnedAboutTSVersion = false;
 
-export function warnAboutTSVersion(parseSettings: ParseSettings): void {
+export function warnAboutTSVersion(
+  parseSettings: ParseSettings,
+  passedLoggerFn: boolean,
+): void {
   if (!isRunningSupportedTypeScriptVersion && !warnedAboutTSVersion) {
-    const isTTY =
-      typeof process === 'undefined' ? false : process.stdout?.isTTY;
-    if (isTTY) {
-      const border = '=============';
-      const versionWarning = [
-        border,
-        'WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.',
-        'You may find that it works just fine, or you may not.',
-        `SUPPORTED TYPESCRIPT VERSIONS: ${SUPPORTED_TYPESCRIPT_VERSIONS}`,
-        `YOUR TYPESCRIPT VERSION: ${ACTIVE_TYPESCRIPT_VERSION}`,
-        'Please only submit bug reports when using the officially supported version.',
-        border,
-      ];
-      parseSettings.log(versionWarning.join('\n\n'));
+    const border = '=============';
+    const versionWarning = [
+      border,
+      'WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.',
+      'You may find that it works just fine, or you may not.',
+      `SUPPORTED TYPESCRIPT VERSIONS: ${SUPPORTED_TYPESCRIPT_VERSIONS}`,
+      `YOUR TYPESCRIPT VERSION: ${ACTIVE_TYPESCRIPT_VERSION}`,
+      'Please only submit bug reports when using the officially supported version.',
+      border,
+    ].join('\n\n');
+
+    if (passedLoggerFn) {
+      parseSettings.log(versionWarning);
+    } else {
+      const isTTY =
+        typeof process === 'undefined' ? false : process.stdout?.isTTY;
+      if (isTTY) {
+        parseSettings.log(versionWarning);
+      }
     }
```

Comment:
[Cleanup] Nitpicking: could reduce a bit of code here and not initialize the variables if we don't need, thoughts?

```suggestion
    if (
      passedLoggerFn ||
      (typeof process !== 'undefined' && process.stdout?.isTTY)
    ) {
      const border = // ...
      const versionWarning = // ...
      parseSettings.log(versionWarning);
    }
```

I also kind of like reducing nesting of the function by starting off with:

```ts
if (isRunningSupportedTypeScriptVersion || warnedAboutTSVersion) {
  return;
}
```

(this is all nitpicking, no worries if you hate it and don't want it)

Reply:
I applied your suggestions! I think they read better